docs/architecture/adr-008-blocksize-independent-commitment.md

```diff
@@ -0,0 +1,98 @@
+
+![Old Commitment](./assets/size-dependent-commitment.png)
+
+To have a commitment independent of the block size, you would need to create a merkle tree over subroots that are included in every possible block size.
+If we agree on producing the commitment in the `minimalBlocksize` the resulting commitment will be the same no matter the block size. *This is because all the subroots in the `minimalBlocksize` are also included in bigger block sizes*.
+You can see this in the updated version of the diagram above. Now the merkle root will stay the same. This property results from the block width and length being a power of 2.  
+
+![New Commitment](./assets/size-independent-commitment.png)
+
+You can see in the Diagramm below that no matter how the message is split up into rows, the hashes of the subroots stay the same.
+
+![Row Size Comparison](./assets/row-size-comparison.png)
+
+This scheme works for interactive commitments as well as long as the index of when a message starts is the same in every block size.
+For example, in the following diagram, you see the message starting at the second share and still having the same commitment. I marked the skipped blocks pink to show that both messages have the same starting index.
+
+![Interactive Commitment](./assets/interactive-commitment.png)
+
+If the message starting point index is larger than the row of `minimalBlocksize`, you take the index mod `minimalBlocksize`. The `minimalBlocksize` is shown in green and the index in pink. In both cases, the commitment stays the same.
+
+![Interactive Commitment 2](./assets/interactive-commitment2.png)
+
+## Alternative Approaches
+
+TODO
+
+## Decision
+
+TODO
+
+## Detailed Design
+
+To implement this decision, you need to change [`CreateCommit`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L112-166).
+In Detail, [`powerOf2MountainRange`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L142) should take `minimalBlocksize` as an argument instead of `squareSize`.
+
+Following the non-interactive default rules, `minimalBlocksize` can be calculated like this:
+
+```go
+func minimalBlocksize(mLength uint64) uint64 {
+    blocksize := NextHigherPowerOf2(mLength)
+    //Check if message fits with non-interactive-deafult rules
+    if mLength <= (blocksize * (blocksize -1)) {
+        return blocksize
+    } else {
+        return blocksize << 1
+    }
+}
+```
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Negative
+
+1. The amount of subtroots per commitment increases from O(log(n)) to O(sqrt(n)) while n is the number of shares of a message. The worst case for the most amount of shares is depicted in the diagram below - an entire block missing one share.  
+  ![Interactive Commitment 2](./assets/complexity.png)
```

Comment:
[question] in this diagram, I think the y-axis `O(sqrt(n))` is the number of subtree roots. What is the x-axis `O(log(sqrt(n))`?

Reply:
The x-axis is also the number of subtree roots. Here we go after the mountainRange Merkle Tree. So in this case the complexity is logarithmic to the size of the row (sqrt(n)). In this example it would be a maximum of 3 subtree roots.

Reply:
Sorry I don't follow. What does the Merkle mountain range look like for this example and why are there only 3 subtree roots in it?

Reply:
I added the diagram above as they are the same example. The 3 subtree roots are A B C in this case.